pkg/pillar/worker/worker_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// worker is used to kick off some work to a goroutine and get a notification
+// when the work is complete
+
+package worker
+
+import (
+	"errors"
+	"github.com/stretchr/testify/assert"
+	"log"
+	"testing"
+	"time"
+)
+
+var timestamp time.Time
+
+func TestWork(t *testing.T) {
+	testMatrix := map[string]struct {
+		description dummyDescription
+	}{
+		"output": {
+			description: dummyDescription{
+				generateOutput: "test1",
+			},
+		},
+		"output + sleep": {
+			description: dummyDescription{
+				sleepTime:      1,
+				generateOutput: "test2",
+			},
+		},
+		"output + error": {
+			description: dummyDescription{
+				generateOutput: "test3",
+				generateError:  true,
+			},
+		},
+		"output + sleep + error": {
+			description: dummyDescription{
+				sleepTime:      1,
+				generateOutput: "test4",
+				generateError:  true,
+			},
+		},
+	}
+	ctx := dummyContext{contextName: "testContext"}
+	worker := NewWorker(dummyWorker, &ctx, 1)
+	for testname, test := range testMatrix {
+		t.Logf("Running test case %s", testname)
```

Comment:
You may want to use t.Run here to clearly indicate individual testcases

Reply:
done